### ISSUE TITLE
CNN improvements

### ISSUE THREAD

**Issue by Greninja34** [acmpesuecc/pokedex#4]
## CNN model improvement

The current CNN model has only a single convolutional layer, resulting in poor classification accuracy. To significantly improve performance, the model needs a deeper architecture and must be trained on a larger dataset containing diverse images of all 151 first-generation Pokémon.

All changes, experiments, and training code must be implemented strictly within the model_training.py file.

### Goals

1. Design and implement a deeper CNN model architecture with multiple convolutional, pooling, and dense layers.
2. Train and evaluate the model on a comprehensive image dataset of the first-generation Pokémon.
3. Document the model architecture, training process, evaluation metrics, and visual results.

### Tasks

* Research and download a dataset containing images of all 151 first-generation Pokémon.
  * Aim for around 50–100 images per Pokémon (approximately 7,000–15,000 images total).
  * Ensure dataset is well-balanced and labeled correctly.
* Improve the CNN model with additional convolutional, pooling, and fully connected layers.
* Train the model on GPU-enabled hardware (local system with GPU, macOS, Google Colab, or Kaggle).
* Visualize and document the final CNN architecture (export as an image).
* Plot and include graphs for:
  * Training loss vs. epochs
  * Training accuracy vs. epochs
  * Validation loss vs. epochs
  * Validation accuracy vs. epochs
* Report final model accuracy, loss, and validation metrics.
* Include several test case predictions with images and their predicted labels.
* Create a documentation to include:
  * Dataset details (source, structure, preprocessing)
  * Model architecture summary
  * Training configurations and hyperparameters
  * Evaluation results and visualizations

**Comment by Arjun9514** [comment]
Can i work on this? @Greninja34

**Comment by SiddarthAA** [comment]
hi @Greninja34 can you **please** assign this to me thanks!!

**Comment by Greninja34** [comment]
!assign @Arjun9514 60

**Comment by Greninja34** [comment]
!assign @SiddarthAA 60

**Comment by SiddarthAA** [comment]
can you try again @Greninja34 ive withdrawn from the other one! thanks

**Comment by Greninja34** [comment]
okay, just so u understand that this will take a lot of time to trian this model, and u need to find an appropriate dataset to train the CNN, as well as i need a high accuracy (~95+), and i need proof of the training, as well as the mathematical graphs of the loss and acc (refer the isssue properly), understood??

tell me which hall you're in, i need to come ask u some questions now

**Comment by SiddarthAA** [comment]
ill probably run inference on colab and then migrate the architecture onto my local system with clean docs! 
sure im in the sem hall 2 last row next to the door!!

**Comment by Greninja34** [comment]
!assign @SiddarthAA 60

**Comment by aamer236** [comment]
can i work on this?

**Comment by sharatchandra-ts** [comment]
Can i work on this

**Comment by SiddarthAA** [comment]
@Greninja34 can you come look at the progress!

**Comment by Greninja34** [comment]
!extend 60

**Comment by SiddarthAA** [comment]
hi the model is training rn, ive made the architecture better + gotten better hyper parameters and better optimizer (ill lyk the results)

**Comment by sanchita-suni** [comment]
can i work on this?

**Comment by Greninja34** [comment]
!deassign
